### PR TITLE
Read Coils example for TCP Server

### PR DESCRIPTION
**Pull request by AnthonyMujic**
Made a series of changes to get the SimpleServer and ReadCoils examples working.
This is my first ever pull request, so I welcome any feedback. 
Not sure if this pull request is too big, and should be split into a number of pull requests, or if it is too basic, and more features should be added before making a pull request?

### CODE REVIEW

File: test/tcp-server-response-handler.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      0x00, 0x01, // transaction id
+      0x00, 0x00, // protocol
+      0x00, 0x04, // byte count
+      0x01,       // unit id
+      0x01,       // function code
+      0x01,       // byte count
+      0x01        // coils
+    ])
+
+    assert(payload = responseBuffer)
```
Comment (stefanpoeter) [inline test/tcp-server-response-handler.test.js]: I don't think that this is working, should be ===

Can one compare buffers like this? `assert(payload.equals(responseBuffer))`

**Comment by stefanpoeter** [comment]
Hey @AnthonyMujic,
great work. I think the way you did it is just fine since these are some basic changes that are related to the ReadCoils Example. I like that you provided a test method.

I'll made a comment on the `assert` in the test, besides that I could merge it.

**Comment by AnthonyMujic** [comment]
Thanks for the feedback. I have corrected the assertion as suggested.

**Comment by stefanpoeter** [comment]
Thanks for your work, keep it up. If you want to continue working on v3 then I suggest I'll make you a project member. What do you say?

**Comment by AnthonyMujic** [comment]
That sounds great.

**Pull request merged**